okhttp-dnsoverhttps/src/test/java/okhttp3/dnsoverhttps/DnsOverHttpsTest.kt

```diff
@@ -167,30 +168,79 @@ class DnsOverHttpsTest {
   @Test
   fun usesCache() {
     val cache = Cache("cache".toPath(), (100 * 1024).toLong(), cacheFs)
     val cachedClient = bootstrapClient.newBuilder().cache(cache).build()
     val cachedDns = buildLocalhost(cachedClient, false)
-    server.enqueue(
-      dnsResponse(
-        "0000818000010003000000000567726170680866616365626f6f6b03636f6d0000010001c00c000500010" +
-          "0000a6d000603617069c012c0300005000100000cde000c04737461720463313072c012c04200010001000" +
-          "0003b00049df00112",
+
+    repeat(2) {
+      server.enqueue(
+        dnsResponse(
+          "0000818000010003000000000567726170680866616365626f6f6b03636f6d0000010001c00c000500010" +
+            "0000a6d000603617069c012c0300005000100000cde000c04737461720463313072c012c04200010001000" +
+            "0003b00049df00112",
+        )
+          .newBuilder()
+          .setHeader("cache-control", "private, max-age=298")
+          .build(),
       )
-        .newBuilder()
-        .setHeader("cache-control", "private, max-age=298")
-        .build(),
-    )
+    }
+
     var result = cachedDns.lookup("google.com")
     assertThat(result).containsExactly(address("157.240.1.18"))
-    val recordedRequest = server.takeRequest()
+    var recordedRequest = server.takeRequest()
     assertThat(recordedRequest.method).isEqualTo("GET")
     assertThat(recordedRequest.path)
       .isEqualTo("/lookup?ct&dns=AAABAAABAAAAAAAABmdvb2dsZQNjb20AAAEAAQ")
+
+    result = cachedDns.lookup("google.com")
+    assertThat(server.takeRequest(1, TimeUnit.MILLISECONDS)).isNull()
+    assertThat(result).isEqualTo(listOf(address("157.240.1.18")))
+
+    result = cachedDns.lookup("www.google.com")
+    assertThat(result).containsExactly(address("157.240.1.18"))
+    recordedRequest = server.takeRequest()
+    assertThat(recordedRequest.method).isEqualTo("GET")
+    assertThat(recordedRequest.path)
+      .isEqualTo("/lookup?ct&dns=AAABAAABAAAAAAAAA3d3dwZnb29nbGUDY29tAAABAAE")
+  }
+
+  @Test
+  fun usesCacheEvenForPost() {
+    val cache = Cache("cache".toPath(), (100 * 1024).toLong(), cacheFs)
+    val cachedClient = bootstrapClient.newBuilder().cache(cache).build()
+    val cachedDns = buildLocalhost(cachedClient, false, post = true)
+    repeat(2) {
+      server.enqueue(
+        dnsResponse(
+          "0000818000010003000000000567726170680866616365626f6f6b03636f6d0000010001c00c000500010" +
+            "0000a6d000603617069c012c0300005000100000cde000c04737461720463313072c012c04200010001000" +
+            "0003b00049df00112",
+        )
+          .newBuilder()
+          .setHeader("cache-control", "private, max-age=298")
+          .build(),
+      )
+    }
+
+    var result = cachedDns.lookup("google.com")
+    assertThat(result).containsExactly(address("157.240.1.18"))
+    var recordedRequest = server.takeRequest()
+    assertThat(recordedRequest.method).isEqualTo("POST")
+    assertThat(recordedRequest.path)
+      .isEqualTo("/lookup?ct")
```

Comment:
Off-topic: we’re naughty for having a field called `path` that contains a path and a query. Thanks for confirming that we use the right URL for these!